CSS tidyup, Table glitch fix, sticky "Newer version" banner

osfameron opened the pull request:
Some reorg work I did while trying to get head around our CSS.
As I want to use this as basis for future work, this branch does now also include 2 features:

* Table captions on https://preview.docs-test.couchbase.com/hakim/server/current/backup-restore/enterprise-backup-restore.html#%23version-compatibility now fixed (via @malarky)
* The "A newer version is available" banner is now sticky, and the scroll-top distance is tweaked to prevent deep links, e.g. https://preview.docs-test.couchbase.com/hakim/server/7.1/tutorials/install-couchbase-server.html#set-up-a-new-cluster from being hidden behind the banner (via @krugster)

NB: this second fix does use dynamic css variables, which means: 

* we have to disable a postcss plugin…
* … which may impact time-to-first-paint or other metrics, to be monitored
* … and will cause some breakage potentially on obsolete browsers like IE11
* … and might have other subtle behaviours from the css variables now being dynamic instead of static

(I think it's worth pushing in any case, and fixing issues as we come across them, but happy to hear concerns, or for 👀 to point out any concrete issues with the implementation!)

osfameron committed:
css-tidyup

osfameron committed:
Remove Contributor box

It's nice to credit contributors...
but doing so in an invisible div that makes an API request
to Github on every page load perhaps not.

osfameron committed:
Reorganize site.css

Refactor callouts.css

base.css - document, purge tailwind

css-tidyup

Comments.
Merge toc.css and toolbar.css

css-tidyup minor components

osfameron committed:
Appeasing css linter

sarahlwelton commented on src/css/base.css:
```diff
@@ -216,218 +213,3 @@ label::after {
-  .mb-lg-5 {
-    margin-bottom: var(--base-large-space) !important;   /* 32px */
-  }
-}
```
Just curious why we deleted all of this spacing CSS?

osfameron replied:
I couldn't find any evidence of it being *used*.
It looks like tailwindCSS convention, which isn't used elsewhere in the project, so I *think* it's an experiment that wasn't furthered.

sarahlwelton commented on src/css/doc.css:
```diff
@@ -509,19 +477,6 @@ ul ul ul {
   font-weight: inherit;
 }
 
-/* Use this for a working pseudo element before an [abstract] attribute-marked block in the documentation.
-  This one adds the words "Quick Summary" before any text inside the [abstract] block. */
-
-/*
-.doc .abstract blockquote::before {
-  content: "Quick Summary \a";
-  white-space: pre;
-  color: var(--color-muted);
-  font-weight: var(--weight-medium);
-  font-size: var(--heading-h4);
-  padding-bottom: 1rem;
-}  */
-
```
You asked me to leave this in, in case we wanted to use it as a working example of a Pseudo :before in the future :) Sure you want to delete it?

osfameron replied:
Hahaha, good catch.
(Trying to purge commented CSS as it adds to cognitive weight, but will have a think...)

sarahlwelton commented on src/css/header.css:
```diff
@@ -208,7 +200,7 @@
 
 .navbar-dropdown li.current a.navbar-item::before {
-  content: "\2023";
+  content: "\2023"; /* ‣ */
```
Is it okay to have a special character like that in here?

osfameron replied:
pretty sure, inside a comment. 
(I suspect it'd be fine inside the quotes too, but haven't tested extensively...)

sarahlwelton commented on src/css/toc.css:
```diff
@@ -129,3 +117,57 @@
   padding-top: 2px;
 }
+
+/* Sidebar Box and Tools */
```
Is this for the Redocly integration, or?

osfameron replied:
Nope, this PR is intended to be a refactor, so it's just decribing the existing sidebar box.

sarahlwelton reviewed (commented):
Couple questions :)

simon-dew commented on src/css/table.css:
I guess you're just tidying up at the moment, but this file still contains the problematic table caption styles?

osfameron replied:
That's right, this PR is a refactor and shouldn't contain any new features (and with a bit of luck, no new breakage either... 😅)

osfameron replied:
(NB: now not true, see the most recent commits)

osfameron committed:
Fix table captions

Glitch reported by @malarky.
Removed special display/position handling.

osfameron committed:
Sticky article-banner

Make article-banner ("A newer version of this document is available")
sticky.
This requires updating the scroll-top, otherwise when clicking on a link
it would be hidden behind the banner.
The existing handling already accounts for the same issue, but with the
height (7.1rem) of the top nav.
We use a dynamic css `var(--scroll-margin-top-space)` which can be
overridden in the case that we have a banner.

sarahlwelton commented on src/partials/article.hbs:
```diff
@@ -5,7 +5,6 @@
 {{> labels}}
 </div>
 {{/if}}
-{{> contributor-bot}}
```
What was contributor-bot? Why are we deleting it?

osfameron replied:
Go to e.g. view-source:https://docs.couchbase.com/server/current/getting-started/do-a-quick-install.html and you'll see `<div class="contributor-list-box">`
Inspect the DOM and you'll see it's been filled with

```
<ul id="contributorList"><li><a href="https://github.com/mojavelinux" target="_blank"><img src="https://avatars.githubusercontent.com/u/79351?v=4" alt=""></a></li><li><a href="https://github.com/cb-docs-robot" target="_blank"><img src="https://avatars.githubusercontent.com/u/25054062?v=4" alt=""></a></li><li><a href="https://github.com/anmol-algo" target="_blank"><img src="https://avatars.githubusercontent.com/u/58175473?v=4" alt=""></a></li><li><a href="https://github.com/RayOffiah" target="_blank"><img src="https://avatars.githubusercontent.com/u/77050471?v=4" alt=""></a></li><li><a href="https://github.com/amarantha-k" target="_blank"><img src="https://avatars.githubusercontent.com/u/13425257?v=4" alt=""></a></li></ul>
```

e.g. it's doing an API request to github stats for docs-ui on every page load to fill this in.

osfameron commented:
(Left too long unmerged, will have to come back to this to see if there's anything that can be introduced -- as smaller PRs...)
